### DISCUSSION TITLE
Geographical location of a scene

### DISCUSSION THREAD

**Discussion by HermannStettin** [azavea/raster-vision#2175] [Q&A] [answered]
### Description
I have a set of relatively small (~200x~200) scenes which are not geo-binded, and only their approximate position on the terrain is known with an accuracy of 1 km. Scenes have 4 channels (red, green, blue, near-infrared). I also have large (10980x10980) Sentinel-2 geo-binded images. All this images depict the same location but for different dates. Scenes are parts (slightly transformed cropped parts) of large images.

### Problem
I want to determine coordinates of four corners of each scene in some substrate coordinate system (e.g. EPSG:4326) based on one of the large Sentinel-2 geo-binded images and ideally to geo-bind (add georeferencing) to the scene.

Has anyone solved a similar problem? Is it possible to use this library to solve this problem? Thank you

**Comment by AdeelH** [comment, marked as answer]
So you have some crops of Sentinel-2 scenes and want to find the crop-coordinates that generated them? That sounds like an interesting problem, but not something that Raster Vision supports out-of-the-box. You might still find Raster Vision useful for reading the data though. 

I would try finding the highest cross-correlation. Check out: 
- https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.correlate2d.html (not efficient for 10Kx10K images)
- or FFT-based convolution: https://stackoverflow.com/questions/62026550/why-does-scipy-signal-correlate2d-use-such-an-inefficient-algorithm
- or OpenCV template matching: https://docs.opencv.org/4.x/d4/dc6/tutorial_py_template_matching.html
